pkg/chartverifier/checks/helm.go

```diff
@@ -200,27 +201,39 @@ func getImageReferences(chartUri string, vals map[string]interface{}) ([]string,
 	mem.SetNamespace("TestNamespace")
 	actionConfig.Releases = storage.Init(mem)
 
-	imagesMap := make(map[string]bool)
-
 	txt, err := actions.RenderManifests("testRelease", chartUri, vals, actionConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	return getImagesFromContent(txt)
+
+}
+
+func getImagesFromContent(content string) ([]string, error) {
+
+	imagesMap := make(map[string]bool)
 
 	type ImageRef struct {
 		Ref string `yaml:"image"`
 	}
 
-	if err == nil {
-		r := strings.NewReader(txt)
-		scanner := bufio.NewScanner(r)
-		for scanner.Scan() {
-			line := scanner.Text()
-			var imageRef ImageRef
-			yamlErr := yaml.Unmarshal([]byte(line), &imageRef)
-			if yamlErr == nil {
-				if len(imageRef.Ref) > 0 && !imagesMap[imageRef.Ref] {
-					imagesMap[imageRef.Ref] = true
-				}
+	r := strings.NewReader(content)
+	reader := bufio.NewReader(r)
+	line, err := getNextLine(reader)
+	for err == nil {
+		var imageRef ImageRef
+		yamlErr := yaml.Unmarshal([]byte(line), &imageRef)
+		if yamlErr == nil {
+			if len(imageRef.Ref) > 0 && !imagesMap[imageRef.Ref] {
+				imagesMap[imageRef.Ref] = true
 			}
 		}
+		line, err = getNextLine(reader)
+		if err == io.EOF {
+			err = nil
+			break
+		}
 	}
```

Comment:
I would rewrite this do-while pattern as the following suggestion:

```suggestion
	for line, err := getNextLine(reader); err != io.EOF {
		if err != nil {
			// an error occured
			break
		} 
		var imageRef ImageRef
		yamlErr := yaml.Unmarshal([]byte(line), &imageRef)
		if yamlErr == nil {
			if len(imageRef.Ref) > 0 && !imagesMap[imageRef.Ref] {
				imagesMap[imageRef.Ref] = true
			}
		}
	}
```

Reply:
You loop has issues, the scope of err is limited to the loop so is not available for return, then if I change the scope of err the loop wold exit with err set to io.EOF so would need setting back to nil.  As a result I left loop as-is as it was the simplest thing to do and I know it works.

Reply:
Let's then rename `getNextLine` error to `gnlErr`, and copy it to the function scoped `err` before break:

```go
// gnlErr will be discarded and the resolution error won't leak to `err`
for line, gnlErr := getNextLine(reader); gnlErr != io.EOF {
        if gnlErr != nil {
                err = gnlErr  // assign the local error to the function scoped `err`
                break
        }
        ...
}
```